diff --git a/JsonGenerator/source/class_emitter.py b/JsonGenerator/source/class_emitter.py
index 271b788..d653c6b 100644
--- a/JsonGenerator/source/class_emitter.py
+++ b/JsonGenerator/source/class_emitter.py
@@ -250,7 +250,6 @@ def EmitEnumRegs(log, root, emit, header_file, if_file):
     emit.Line("#include <core/Enumerate.h>")
     emit.Line()
 
-    emit.Line("#include \"definitions.h\"")
 
     if not config.NO_INCLUDES:
         if if_file.endswith(".h"):
diff --git a/ProxyStubGenerator/StubGenerator.py b/ProxyStubGenerator/StubGenerator.py
index 4b4e3a5..f8323c8 100755
--- a/ProxyStubGenerator/StubGenerator.py
+++ b/ProxyStubGenerator/StubGenerator.py
@@ -201,7 +201,7 @@ def Flatten(identifier, scope):
 
 
 # Generate interface information in lua
-def GenerateLuaData(emit, interfaces_list, enums_list, source_file=None, tree=None, ns=None):
+def GenerateLuaData(emit, interfaces_list, enums_list, project_dir, source_file=None, tree=None, ns=None):
 
     if not source_file:
         assert tree==None
@@ -649,7 +649,7 @@ def Parse(source_file, framework_namespace, includePaths = [], defaults = "", ex
 
     return tree
 
-def GenerateStubs2(output_file, source_file, tree, ns, scan_only=False):
+def GenerateStubs2(output_file, source_file, project_dir, tree, ns, scan_only=False):
     log.Info("Scanning '%s' (in %s)..." % (source_file, ns))
 
     if not FORCE and (os.path.exists(output_file) and (os.path.getmtime(source_file) < os.path.getmtime(output_file))):
@@ -2489,7 +2489,7 @@ def GenerateStubs2(output_file, source_file, tree, ns, scan_only=False):
 
         emit.Line()
 
-        if os.path.isfile(os.path.join(os.path.dirname(source_file), "Module.h")):
+        if os.path.isfile(os.path.join(project_dir, "Module.h")):
             emit.Line('#include "Module.h"')
 
         if os.path.isfile(os.path.join(os.path.dirname(source_file), interface_header_name)):
@@ -2631,6 +2631,8 @@ if __name__ == "__main__":
                            help="include an additional C++ header file, can be used multiple times (default: include 'Module.h')")
     argparser.add_argument('-I', dest="includePaths", metavar="INCLUDE_DIR", action='append', default=[], type=str,
                            help='add an include search path, can be used multiple times')
+    argparser.add_argument("--projectdir",dest="project_dir",metavar="DIR",type=str,default="",
+                           help="specify the project directory")
 
     args = argparser.parse_args(sys.argv[1:])
     SHOW_WARNINGS = not args.no_warnings
@@ -2772,8 +2774,12 @@ if __name__ == "__main__":
 
             for source_file in interface_files:
                 try:
-                    _extra_includes = [ os.path.join("@" + os.path.dirname(source_file), MODULE_FILE) ]
-                    _extra_includes = [ os.path.join("@" + os.path.dirname(source_file), "Ids.h") ]
+                    if args.project_dir is not None:
+                        _extra_includes = [ os.path.join("@" + args.project_dir, MODULE_FILE) ]
+                        _extra_includes = [ os.path.join("@" + args.project_dir, "Ids.h") ]
+                    else:
+                        _extra_includes = [ os.path.join("@" + os.path.dirname(source_file), MODULE_FILE) ]
+                        _extra_includes = [ os.path.join("@" + os.path.dirname(source_file), "Ids.h") ]
 
                     _extra_includes.extend(args.extra_includes)
 
@@ -2795,7 +2801,7 @@ if __name__ == "__main__":
                         some_omitted = False
 
                         for ns in INTERFACE_NAMESPACES:
-                            output, some_omitted = GenerateStubs2(output_file, source_file, tree, ns, scan_only)
+                            output, some_omitted = GenerateStubs2(output_file, source_file, args.project_dir, tree, ns, scan_only)
 
                             new_faces += output
 
@@ -2818,7 +2824,7 @@ if __name__ == "__main__":
                         log.Info("(lua generator) Scanning %s..." % os.path.basename(source_file))
 
                         for ns in INTERFACE_NAMESPACES:
-                            GenerateLuaData(Emitter(lua_file, INDENT_SIZE), lua_interfaces, lua_enums, source_file, tree, ns)
+                            GenerateLuaData(Emitter(lua_file, INDENT_SIZE), lua_interfaces, lua_enums, args.project_dir, source_file, tree, ns)
 
                 except NotModifiedException as err:
                     log.Info("skipped file %s, up-to-date" % os.path.basename(output_file))
@@ -2869,7 +2875,7 @@ if __name__ == "__main__":
             if args.lua_code:
                 # Epilogue
                 for ns in INTERFACE_NAMESPACES:
-                    GenerateLuaData(Emitter(lua_file, INDENT_SIZE), lua_interfaces, lua_enums)
+                    GenerateLuaData(Emitter(lua_file, INDENT_SIZE), lua_interfaces, lua_enums, args.project_dir)
                     log.Info("Created %s (%s interfaces, %s enums)" % (lua_file.name, len(lua_interfaces), len(lua_enums)))
 
         else:
diff --git a/cmake/FindProxyStubGenerator.cmake.in b/cmake/FindProxyStubGenerator.cmake.in
index e6e2709..355ae4d 100644
--- a/cmake/FindProxyStubGenerator.cmake.in
+++ b/cmake/FindProxyStubGenerator.cmake.in
@@ -31,7 +31,7 @@ function(ProxyStubGenerator)
     endif()
 
     set(optionsArgs SECURE COHERENT TRACES VERBOSE NO_WARNINGS KEEP FORCE_GENERATE)
-    set(oneValueArgs OUTDIR)
+    set(oneValueArgs OUTDIR PROJECT_DIR)
     set(multiValueArgs INPUT INCLUDE INCLUDE_PATH NAMESPACE)
 
     cmake_parse_arguments(Argument "${optionsArgs}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
@@ -81,6 +81,10 @@ function(ProxyStubGenerator)
         list(APPEND _execute_command  "--outdir" "${Argument_OUTDIR}")
     endif()
 
+    if(Argument_PROJECT_DIR)
+       list(APPEND _execute_command  "--projectdir" "${Argument_PROJECT_DIR}")
+    endif()
+
     foreach(_include ${Argument_INCLUDE})
         list(APPEND _execute_command  "-I" "${_include}")
     endforeach(_include)
