From: rkiran696_comcast <raja_kiran@comcast.com>
Date: Jul 29, 2026
Source: Comcast

diff --git a/Source/core/Frame.h b/Source/core/Frame.h
index dd9c8368..e751efdd 100644
--- a/Source/core/Frame.h
+++ b/Source/core/Frame.h
@@ -218,6 +218,15 @@ namespace Core {
             {
                 return (_container == nullptr ? 0 : _container->Size() - _offset);
             }
+            SIZE_CONTEXT LockFixedBuffer(const uint8_t*& buffer, const uint32_t length) const
+            {
+                ASSERT(_container != nullptr);
+                ASSERT(length <= Length());
+
+                buffer = &(_container->operator[](_offset));
+
+                return (length < Length() ? length : Length());
+            }
             template <typename TYPENAME>
             TYPENAME LockBuffer(const uint8_t*& buffer) const
             {
