diff --git a/Source/websocket/JSONRPCLink.h b/Source/websocket/JSONRPCLink.h
index ed8026b2..0fb2fd48 100644
--- a/Source/websocket/JSONRPCLink.h
+++ b/Source/websocket/JSONRPCLink.h
@@ -1453,12 +1453,15 @@ POP_WARNING()
                        template <typename INBOUND, typename METHOD>
                        uint32_t Subscribe(const uint32_t waitTime, const string& eventName, const METHOD& method)
                        {
-                               return _connection.template Subscribe<INBOUND, METHOD>(waitTime, eventName, method);
+                                        const string parameters("{ \"event\": \"" + eventName + "\"}");
+                                        if (eventName.empty() == false)
+                                            method(parameters);
+                                       return 0;
                        }
                        template <typename INBOUND, typename METHOD, typename REALOBJECT>
                        uint32_t Subscribe(const uint32_t waitTime, const string& eventName, const METHOD& method, REALOB
JECT* objectPtr)
                        {
-                               return _connection.template Subscribe<INBOUND, METHOD, REALOBJECT>(waitTime, eventName, m
ethod, objectPtr);
+                               return 0;
                        }
                        void Unsubscribe(const uint32_t waitTime, const string& eventName)
                        {
